app/src/main/java/com/mailerdaemon/app/LoginActivity.kt

```diff
@@ -43,31 +39,26 @@ class LoginActivity: AppCompatActivity() {
         if (getSharedPreferences("MAIN", Context.MODE_PRIVATE).getBoolean("intro", true)) {
             startActivity(Intent(this, IntroActivity::class.java))
             finish()
-        } else {
-            val currentUser = mAuth.currentUser
-            currentUser?.let { startMain() }
-        }
+        } else
+            startMain()
         setContentView(R.layout.activity_login)
         progress_bar.visibility = View.GONE
         forgot_password.setOnClickListener { startActivity(Intent(this, ForgotPassActivity::class.java)) }
         login.setOnClickListener {
-            val email = Objects.requireNonNull(login_email.text).toString().trim { it <= ' ' }
-            val password = Objects.requireNonNull(login_password.text).toString()
+            val email = login_email!!.text.toString().trim { it <= ' ' }
+            val password = (login_password!!.text).toString()
```

Comment:
remove `!!` use `val email = expression?.xyz ?: "" ` this will make email empty when it is null.